main.md

```diff
@@ -401,6 +435,91 @@ known type but containing unknown elements or elements of the wrong type, the AS
+
+## Comparing authorization details
+
+Many actions in the OAuth protocol allow the AS and RS to make security decisions based on whether or not the request
+is asking for "more" or "less" than a previous, existing request. For example, upon refreshing a token, the client can
+ask for a new access token with "fewer permissions" than had been previously authorized by the resource owner.
+Since the nature of an authorization details request is based solely on the API or APIs that it is describing, there is not
+a simple means of comparing any two arbitrary authorization details requests. 
+Authorization servers should not rely on simple object comparison in most cases, as the intersection of some elements
+within a request could have side effects in the access rights granted, depending on how the API
+has been designed and deployed. This is a similar effect to the scope values used with some APIs.
+
+However, when comparing a new request to an existing request, authorization servers can use the same 
+processing techniques as used in granting the request in the first place to determine if a resource
+owner needs to authorize the request. The details of this comparison are dependent on the definition
+of the `type` of authorization request and outside the scope of this specification, but common patterns
+can be applied.
+
+This shall be illustrated using our running example. The example authorization request in (#authz_request), if approved by the user, resulted in the issuance of an authorization code associated with the privileges to 
+
+* list accounts 
+* access the balance of one or more accounts,
+* access the transactions of one or more accounts, and 
+* to initiate a payment. 
+
+The client could now request the AS to issue an access token assigned with the privilege to just access a list of accounts as follows:
+
+```JSON
+[
+   {
+      "type":"account_information",
+      "actions":[
+         "list_accounts"
+      ],
+      "locations":[
+         "https://example.com/accounts"
+      ]
+   }
+]
+```
+
+The example API is designed such that each field used by the `account_information` type contains additive rights, 
+with each value within the `actions` and `locations` arrays specifying a different element of access. To make a comparison in this
+instance, the AS would perform the following steps:
+
+* compare that the authorization code issued in the previous step contains an authorization details object of type `account_information`
+* compare whether the approved list of actions contains `list_account`, and 
+* whether the `locations` value includes only previously-approved locations.
+
+If all checks succeed, the AS would issue the requested access token with the reduced set of access. 
+
+Note that this comparison is relevant A different API type definition could have different processing rules. For example, the
```

Comment:
`Note that this comparison is relevant A different API type definition could have different processing rules.` something isn't right here. `Note that this comparison is relative. A different API type definition could have different processing rules.` maybe? I dunno.

Reply:
My bad, lost some text:

```suggestion
Note that this comparison is relevant to this specific API type definition. A different API type definition could have different processing rules. For example, the
```

Reply:
better!